js/main.js

```diff
@@ -29,29 +29,49 @@ myApp.controller('username', ['$scope', '$window', function($scope, $window) { /
 // custom states (pages)
 myApp.config(['$stateProvider', require('./segments/segmentsState')]);
 
-myApp.config(['NgAdminConfigurationProvider', function (nga) {
+myApp.config(['NgAdminConfigurationProvider', '$windowProvider', function (nga, $window) {
     // create the admin application
     var admin = nga.application('My First Admin')
         .baseApiUrl('/');
 
-    // add entities
-    admin.addEntity(nga.entity('customers'));
-    admin.addEntity(nga.entity('categories'));
-    admin.addEntity(nga.entity('products'));
-    admin.addEntity(nga.entity('reviews'));
-    admin.addEntity(nga.entity('commands'));
-
-    // configure entities
-    require('./customers/config')(nga, admin);
-    require('./categories/config')(nga, admin);
-    require('./products/config')(nga, admin);
-    require('./reviews/config')(nga, admin);
-    require('./commands/config')(nga, admin);
-
-    admin.dashboard(require('./dashboard/config')(nga, admin));
-    admin.header(require('./header.html'));
-    admin.menu(require('./menu')(nga, admin));
-
-    // attach the admin application to the DOM and execute it
-    nga.configure(admin);
+    if ($window.$get().localStorage.getItem('posters_galore_login') == 'admin') { //retrieving current role
+        // add entities
+        admin.addEntity(nga.entity('customers'));
+        admin.addEntity(nga.entity('categories'));
+        admin.addEntity(nga.entity('products'));
+        admin.addEntity(nga.entity('reviews'));
+        admin.addEntity(nga.entity('commands'));
+
+        // configure entities
+        require('./customers/config')(nga, admin);
+        require('./categories/config')(nga, admin);
+        require('./products/config')(nga, admin);
+        require('./reviews/config')(nga, admin);
+        require('./commands/config')(nga, admin);
+
+        admin.dashboard(require('./dashboard/config')(nga, admin));
+        admin.header(require('./header.html'));
+        admin.menu(require('./menu')(nga, admin));
+
+        // attach the admin application to the DOM and execute it
+        nga.configure(admin);
+    } else {
+        // add entities
+        admin.addEntity(nga.entity('customers').readOnly());
+        admin.addEntity(nga.entity('categories').readOnly());
+        admin.addEntity(nga.entity('products').readOnly());
+
+        // configure entities
+        require('./customers/userConfig')(nga, admin);
```

Comment:
I'm saying that you shouldn't put the `if` here, but in the config files. Something like:

```js
const roles = $window.$get().localStorage.getItem('posters_galore_roles')
admin.addEntity(nga.entity('customers'));
require('./customers/config')(nga, admin, roles);
```

And then, in the config files, decide what to add according to roles.